src/Blocks/BlockCustomData.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * BlockCustomData integration class.
+ *
+ * @package EightshiftForms\Blocks
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Blocks;
+
+use EightshiftForms\Form\AbstractFormBuilder;
+use EightshiftForms\Hooks\Filters;
+use EightshiftFormsVendor\EightshiftLibs\Services\ServiceInterface;
+
+/**
+ * BlockCustomData integration class.
+ */
+class BlockCustomData extends AbstractFormBuilder implements ServiceInterface
+{
+	/**
+	 * Filter return custom data component from filter.
+	 *
+	 * @var string
+	 */
+	public const FILTER_BLOCK_CUSTOM_DATA_COMPONENT_NAME = 'es_forms_block_custom_data_component';
+
+	/**
+	 * Register all the hooks
+	 *
+	 * @return void
+	 */
+	public function register(): void
+	{
+		// Blocks string to value filter name constant.
+		\add_filter(static::FILTER_BLOCK_CUSTOM_DATA_COMPONENT_NAME, [$this, 'getCustomDataComponent'], 50);
+	}
+
+	/**
+	 * Build block from Custom data.
+	 *
+	 * @param array<string, mixed> $attributes Block attributes.
+	 *
+	 * @return string
+	 */
+	public function getCustomDataComponent(array $attributes): string
+	{
+		if (!has_filter(Filters::FILTER_BLOCK_CUSTOM_DATA_OPTIONS_DATA_NAME)) {
+			return '';
+		}
+
+		$customDataData = $attributes['customDataData'] ?? '';
+
+		$data = apply_filters(Filters::FILTER_BLOCK_CUSTOM_DATA_OPTIONS_DATA_NAME, $customDataData);
+
+		if (!$data) {
+			return '';
+		}
+
+		$customDataFieldType = $attributes['customDataFieldType'] ?? '';
+
+		switch ($customDataFieldType) {
+			case 'checkboxes':
+				$output = [
+					'component' => 'checkboxes',
+					'checkboxesFieldLabel' => $attributes['customDataCheckboxesFieldLabel'] ?? '',
+					'checkboxesId' => $attributes['customDataId'] ?? '',
+					'checkboxesName' => $attributes['customDataCheckboxesName'] ?? '',
+					'checkboxesContent' => array_map(
+						function ($option) {
+							return [
+								'component' => 'checkbox',
+								'checkboxLabel' => $option['label'] ?? '',
+								'checkboxValue' => $option['value'] ?? $option['label'] ?? '',
+								'checkboxIsChecked' => $option['selected'] ?? false,
+							];
+						},
```

Comment:
I haven't got the time to comment, but you can open a new PR just for this.

Wherever you are using anonymous functions (like here in `array_map`) if you are not using `$this` keyword in a function (callback) you can improve the performance by adding a `static` in front of the `function`.

Like:

```php
'checkboxesContent' => array_map(
	static function ($option) {
		return [
			'component' => 'checkbox',
			'checkboxLabel' => $option['label'] ?? '',
			'checkboxValue' => $option['value'] ?? $option['label'] ?? '',
			'checkboxIsChecked' => $option['selected'] ?? false,
		];
	},
	$data
),
```

Reply:
awesome did't know that. I have opened a issue tnx